cmake-tool/helpers/environment_flags.cmake

```diff
@@ -92,6 +92,15 @@ macro(find_libgcc_files)
     # find the compilers crtbegin and crtend files
     gcc_print_file_name(CRTBeginFile crtbegin.o)
     gcc_print_file_name(CRTEndFile crtend.o)
+
+    # Empty crt files variables if not found in the current toolchain.
+    # This should allow building/linking without cxx crt files (e.g., just C projects).
```

Comment:
Can it happen that just one file of them exists? In this case we also drop it, should we keep it then?

Reply:
I don't think so, they usually have to be bundled together for instructors/destructors

Reply:
The docs say it output the absolute name of the file. It does not say what we get when the file does not exist. Looking at `gcc_print_file_name()` is seem it prints the plain input when the file is not found, but does not check the return code actually.. Have you checked if we we get an error return code? Then we could have `gcc_print_file_name()` log a more detailed warning message and set the variable to empty. That seems be a bit cleaner than doing it here. It might have some side effects on the other places where this is used, but I assume all they run into the same issues basically if the input files does not exists so making the variable empty seem a sane choice.

Reply:
> The docs say it output the absolute name of the file. It does not say what we get when the file does not exist. Looking at `gcc_print_file_name()` is seem it prints the plain input when the file is not found, but does not check the return code actually.. Have you checked if we we get an error return code? Then we could have `gcc_print_file_name()` log a more detailed warning message and set the variable to empty. That seems be a bit cleaner than doing it here. It might have some side effects on the other places where this is used, but I assume all they run into the same issues basically if the input files does not exists so making the variable empty seem a sane choice.

Yeah it just prints `crtend.o` or `crtbegin.o` if the file doesn't exist, with a success (0) return code, unfortunately

Reply:
> Yeah it just prints `crtend.o` or `crtbegin.o` if the file doesn't exist, with a success (0) return code, unfortunately

So, should we still try to handle things within `gcc_print_file_name()` then and make the variable empty instead of returning the input value. It would be a bigger change, but it make the general behavior more sane then for the caller.

Reply:
I think it's the caller responsibility to handle such case in line with GCC/LLVM behaviour, as with anyone calling GCC/LLVM with `--print-file-name` argument, otherwise `--print-file-name` would have returned an error code in the first place. For instance, if the caller is a C++ project and must expect these files, it should fail the build, rather than set these files empty